### PR TITLE
[auick_action_ios] Retries multiple times to not fail ci when there is a flake

### PR DESCRIPTION
**Pull request by cyanglaz**
The only way that we found to test quick action menu is to long press for a x second. Previously, we pressed it for 2 seconds but it is sometimes too long and the quick action menu is disappeared.

In this PR, we:
1. Reduce the press time to 1.5 seconds.
2. Retry 3 times.

fixes https://github.com/flutter/flutter/issues/125509

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [relevant style guides] and ran the auto-formatter. (Unlike the flutter/flutter repo, the flutter/packages repo does use `dart format`.)
- [x] I signed the [CLA].
- [x] The title of the PR starts with the name of the package surrounded by square brackets, e.g. `[shared_preferences]`
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated `pubspec.yaml` with an appropriate new version according to the [pub versioning philosophy], or this PR is [exempt from version changes].
- [ ] I updated `CHANGELOG.md` to add a description of the change, [following repository CHANGELOG style].
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[relevant style guides]: https://github.com/flutter/packages/blob/main/CONTRIBUTING.md#style
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[exempt from version changes]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#version-and-changelog-updates
[following repository CHANGELOG style]: https://github.com/flutter/flutter/wiki/Contributing-to-Plugins-and-Packages#changelog-style
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests

### CODE REVIEW

File: packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift [hunk 1]
```diff
@@ -5,6 +5,9 @@
 import XCTest
 
 private let elementWaitingTime: TimeInterval = 30
+private let quickActionPressDuration: TimeInterval = 1.5
+// Max number of tries to open the quick action menu if failed.
```
Comment (hellohuanlin) [inline packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift:9]: maybe add a short explanation why we need to retry?

File: packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift [hunk 2]
```diff
@@ -73,16 +83,23 @@ class RunnerUITests: XCTestCase {
       )
     }
 
-    quickActionsAppIcon.press(forDuration: 2)
-
-    let actionOne = springboard.buttons["Action one"]
-    if !actionOne.waitForExistence(timeout: elementWaitingTime) {
+    var actionOne: XCUIElement?
+    for _ in 1...quickActionMaxRetries {
```
Comment (hellohuanlin) [inline packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift:87]: may wanna write a helper function?

**Comment by cyanglaz** [comment]
Offline discussion: Maybe start with 2 seconds and reduce the press duration over the following retires. 
2 seconds the first try, 1.8 seconds the next try, 1.6 seconds the next try etc.

**Comment by hellohuanlin** [comment]
yeah that's probably better

File: packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift (hunk 2 above)
Comment (cyanglaz) [reply packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift:91]: done

File: packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift (hunk 1 above)
Comment (cyanglaz) [reply packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift:17]: done

File: packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift [hunk 3]
```diff
@@ -73,24 +73,50 @@ class RunnerUITests: XCTestCase {
...
+        // find the button, exit the retry loop.
+        break
+      }
+      let deleteButton = springboard.buttons["DeleteButton"]
+      if deleteButton.waitForExistence(timeout: elementWaitingTime) {
+        // Found delete button instead, we pressed too long, reduce the press time.
+        pressDuration -= pressDurationRetryAdjustment
+      } else {
+        // Neither action button nor delete button was found, we need a longer press.
+        pressDuration += pressDurationRetryAdjustment
```
Comment (hellohuanlin) [inline packages/quick_actions/quick_actions_ios/example/ios/RunnerUITests/RunnerUITests.swift:109]: This makes sense to me. though the drawback is that it makes the test a bit complicated, as test code itself is not tested. i would simply try like 1, 1.33, 1.66 and 2. but up to you.

**Review by hellohuanlin — APPROVED** [review]

**Pull request merged**